action.yml

```diff
@@ -0,0 +1,295 @@
+    default: "balena/testbot-rig"
+  BALENACLOUD_API_URL:
+    required: false
+    default: "balena-cloud.com"
+  BALENAMACHINE_API_URL:
+    required: false
+    default: "bm.balena-dev.com"
+  BALENACLOUD_SSH_URL:
+    required: false
+    default: "ssh.devices.bm.balena-dev.com"
+  BALENACLOUD_SSH_PORT:
+    default: 222
+    required: false
+  QEMU_CPUS:
+    required: false
+    default: 4
+  QEMU_MEMORY:
+    required: false
+    default: "1G"
+  WORKER_TYPE:
+    description: "Valid choices: testbot/qemu"
+    required: true
+  LEVIATHAN_DEBUG:
+    description: |
+      Use this parameter to modify the default debugging settings on test runs. To debug test runs when rebuilding, add new values to this parameter separated with commas. Know more about the possible values available for debugging: https://balena-os.github.io/leviathan/pages/Getting-Started/debugging.html Example value: dev: true, failFast: false
+    required: false
+
+
+outputs:
+  reports:
+    description: "REPORT!"
+    value: ${{ steps.random-number-generator.outputs.random-number }}
+
+runs:
+  using: "composite"
+  steps:
+    # - name: Past Cleanup probably not required
+    #   shell: bash
+    #   run: |
+    #     # unintended consequences
+    #     # It does delete a ton of images on GHR: 
+    #     # Total reclaimed space: 3.141GB
+    #     # docker system prune -a -f || true
+    #     printenv | sort
+
+    - name: Checkout the current repo
+      uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Fetching the right leviathan
+      uses: actions/checkout@v4
+      with:
+        repository: balena-os/leviathan
+        submodules: true
+        path: ./leviathan-workspace
+        # The branch, tag or SHA to checkout. When checking out the repository that
+        # triggered a workflow, this defaults to the reference or SHA for that event.
+        # Otherwise, uses the default branch.
+        ref: ${{ inputs.LEVIATHAN_REF }}
+
+    # - name: Fetching the Private Contracts
+    #   uses: actions/checkout@v4
+    #   with:
+    #     repository: balena-io/private-contracts
+    #     path: './leviathan/core/private-contracts'
+
+
+      # shell: bash
+      # env:
+      #   LEVIATHAN_REF: ${{ inputs.LEVIATHAN_REF }}
+      # run: |
+      #   git clone --recursive https://github.com/balena-os/leviathan leviathan-new/
+      #   cd leviathan-new/
+      #   git fetch origin
+      #   git checkout --progress --force $LEVIATHAN_REF
+
+      #   # NEED KEYS to access private contracts
+      #   # git clone git@github.com:balena-io/private-contracts.git core/private-contracts
+        
+
+    - name: All the copying stage
+      shell: bash
+      working-directory: ./leviathan-workspace
+      run: |
+        mkdir -p ${{ inputs.WORKSPACE }}
+        (cd ${{ inputs.WORKSPACE }} && mkdir -p ${{ inputs.REPORTS }})
+        cp ../${{ inputs.SUITES }}/config.js ${{ inputs.WORKSPACE }}
+        cp -r ../${{ inputs.SUITES }} . 
```

Comment:
The calling workflow should provide these file paths as inputs if not already in the expected locations.

Reply:
These inputs need to be absolute paths from now. I can work on my end to make sure Leviathan can handle any path as long as its valid. How about that?

Reply:
I wouldn't prioritize that, but it might be nice eventually.

IMO as long as the action description and readme are clear about what files/dirs are expected where, we can make sure that the calling workflow has everything in place. They don't need to be inputs for MVP.

AFAIK all steps allow the `working-directory` top-level parameter which should control the root of where the action is run. Let's try using that for now and avoid any inputs for paths and directories.